### DISCUSSION TITLE
Can I update `pyproject.toml` via CLI?

### DISCUSSION THREAD

**Discussion by kingyue737** [python-poetry/poetry#6346] [Q&A] [answered]
For frontend, with `pnpm update`, both `package.json` and `pnpm-lock.yaml` are updated. When I tried `poetry update`, I found that only `poetry.lock` was updated.

**Comment by finswimmer** [comment, marked as answer]
Hey,

`poetry update` updates the version of all dependencies within the given range of the `pyproject.toml`. So it updates the `poetry.lock` file and the environment.

You can upgrade to the latest version via `poetry add <packagename>@latest`. This will bump the version in the `pyproject.toml` as well.

At the moment there is no command to upgrade all packages. A feature request exists: https://github.com/python-poetry/poetry/issues/461

fin swimmer

**Comment by KBeDevel** [comment]
Additionally to the accepted answer, there is a plugin to support this:

https://github.com/apoclyps/poetry-plugin-upgrade
